### DISCUSSION TITLE
using oidc grant access to specific app automatically

### DISCUSSION THREAD

**Discussion by lcslouis** [Budibase/budibase#3559] [Help]
Hello I have search the guides and can't find this information.

I am using keycloak (oidc) as my authentication which i have working user gets created in budibase.

also a bug i noticed is that the auth providers even when blank show enabled and setup doesn't increase its setup progress until both are configured. I had to put non-sense data in the google one and clicked it to disable to have auth be checked off as done. 

Issue is that app role assignment is still manual. how can i configure it to auto give access to an app and a particular role in that app?

Not having any clear documentation on this part makes this difficult.

**Comment by mjashanks** [comment]
Hi there. Thanks for the Bug report :)

There is no way to automatically assign roles at the moment. We expect this to be covered by "User Groups" - see here: https://github.com/Budibase/budibase/discussions/2185

We are not sure when this will come. When it does, it will most likely be a paid (not open-source) feature.

**Reply by lcslouis** [reply]
Handling groups is one thing but when using an auth provider group management is not handled by budibase. 

For open source self hosts external auth needs to be removed because it does not make sense to have external auth and it still be 100% manual app assignment. Because why would I use an external auth if I have to assign every person manually. The answer is I wouldn't because it is doubling my time on adding a user. So remove external auth it should only be a paid feature. Because group management is a paid feature auth should be as well.

**Reply by mjashanks** [reply]
External auth is still extremely useful as is - you get the security and convenience of single-sign-on.

But I see what you are saying now, you want new user to come along, sign in with SSO and see some default set of apps, with default roles assigned to them.

I think this would be useful and not necessarily a paid feature - I've never thought about it.

Out of interest - how would you expect this to work?

i.e. New user comes along and signs in with SSO. How do we determine ....
1. What apps they have access to?
2. What their initial level of access to these apps are?

I think that this should not be the default behaviour - we would want to default on the side of more secure - where every member of the auth provider automatically has access.

**Reply by lcslouis** [reply]
Each auth provider has a scope / role and that information can be pulled from the provider in the app settings you would just need to have a field where the name of the role from the auth coordinates to the app role. 

So on connect of the login from auth it just reads that role and then adds the user to that app at that level because every app made may have different levels of what's public vs what is user and what is staff. 

So to summarize role from a idp designates what role someone has at the basic level. 

To make things more granular or to group control things then you can have a auth role for each app role. 

But basic access for each should be 1 auth role for each app. There may be some people that have access to multiple apps for ever how many they have then that would be that many roles assigned at the idp level.

**Reply by nishantnaagnihotri** [reply]
@mjashanks
This would really help me with my app configuration.
I am in a blocker situation, because it is infeasible for somebody to give app access to each and every user.

Is there any plan to roll out a solution soon?
